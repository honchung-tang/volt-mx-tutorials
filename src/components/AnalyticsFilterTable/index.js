import Table from 'antd/lib/table';
import React from 'react';
import PropTypes from 'prop-types';
import style from './style.scss';

const AnalyticsFilterTable = ({
  showPagination, columns, data, onRowClick, onChange, loading
}) => (
  <Table
    columns={columns}
    dataSource={data}
    bordered
    className={`${style.dataTable} analyticsTable`}
    onChange={onChange}
    loading={loading}
    pagination={
      showPagination
        ? {
          total: data && data.length ? data.length : 0,
          showTotal: (total, range) => `${range[0]}-${range[1]} of ${total} items`,
          pageSize: 12,
        } : false
    }
    onRow={(record, rowIndex) => ({
      onClick: (event) => {
        onRowClick(record);
      },
    })}
  />
);

AnalyticsFilterTable.propTypes = {
  showPagination: PropTypes.bool,
  onRowClick: PropTypes.func,
  columns: PropTypes.array,
  data: PropTypes.array,
  onChange: PropTypes.func,
  loading: PropTypes.bool,
};

AnalyticsFilterTable.defaultProps = {
  showPagination: false,
  onRowClick: () => {},
  onChange: () => {},
  columns: [],
  data: [],
  loading: false,
};

export default AnalyticsFilterTable;
