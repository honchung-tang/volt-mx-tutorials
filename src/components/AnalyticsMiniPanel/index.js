import React from 'react';
import Col from 'antd/lib/col';
import Row from 'antd/lib/row';
import PropTypes from 'prop-types';
import style from './style.scss';
import getConfig from 'next/config';
const { publicRuntimeConfig } = getConfig();

const getIcon = (name) => {
  switch (name) {
    case 'download':
      return (
        <img
          src={`${publicRuntimeConfig.asset}/static/dist/images/icons/download.png`}
          alt="Analytics icon"
          className={style.analyticsIcon}
        />
      );
    case 'import':
      return (
        <img
          src={`${publicRuntimeConfig.asset}/static/dist/images/icons/import.png`}
          alt="Analytics icon"
          className={style.analyticsIcon}
        />
      );
    case 'component':
      return (
        <img
          src={`${publicRuntimeConfig.asset}/static/dist/images/icons/component.png`}
          alt="Analytics icon"
          className={style.analyticsIcon}
        />
      );
    default:
      return (
        <img
          src={`${publicRuntimeConfig.asset}/static/dist/images/icons/import.png`}
          alt="Analytics icon"
          className={style.analyticsIcon}
        />
      );
  }
};

const AnalyticsMiniPanel = ({ title, count, icon, className }) => (
  <Row className={`${style.base} ${className}`}>
    <Col className={style.titleIcon}>
      <Col span={22}>
        <h2>
          {title}
        </h2>
      </Col>
      <Col span={2}>
        {getIcon(icon)}
      </Col>
    </Col>
    <Col span={24}>
      <h3>
        {count}
      </h3>
    </Col>
  </Row>
);

AnalyticsMiniPanel.propTypes = {
  title: PropTypes.string,
  count: PropTypes.number,
  icon: PropTypes.string,
};

AnalyticsMiniPanel.defaultProps = {
  title: '',
  count: 0,
  icon: '',
};

export default AnalyticsMiniPanel;
