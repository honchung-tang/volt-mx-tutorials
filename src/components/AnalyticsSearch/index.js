import React, { Component } from 'react';
import Button from 'antd/lib/button';
import Input from 'antd/lib/input';
import Col from 'antd/lib/col';
import Modal from 'antd/lib/modal';
import Icon from 'antd/lib/icon';
import style from './style.scss';
import getConfig from 'next/config';
const { publicRuntimeConfig } = getConfig();

class AnalyticsSearch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      keyword: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSearchClick = () => {
    this.handleSubmit(document.getElementById('analyticsSearch').value);
  };

  handleSubmit = (keyword) => {
    const searchTerm = keyword.trim().replace(/\\|\/|  +/g, ' ');
    if (searchTerm.length > 2) {
      this.setState({
        submit: true,
      });
      this.props.filterData(keyword);
    } else {
      Modal.error({
        title: 'Please enter minimum 3 characters',
      });
      return false;
    }
  };

  getColumnSearchProps = dataIndex => ({
    filterDropdown: ({
      setSelectedKeys, selectedKeys, confirm, clearFilters,
    }) => (
      <div style={{ padding: 8 }}>
        <Input
          ref={(node) => {
            this.searchInput = node;
          }}
          placeholder={`Search ${dataIndex}`}
          value={selectedKeys[0]}
          onChange={e => setSelectedKeys(e.target.value ? [e.target.value] : [])}
          onPressEnter={() => this.handleSearch(selectedKeys, confirm)}
          style={{ width: 188, marginBottom: 8, display: 'block' }}
        />
        <Button
          type="primary"
          onClick={() => this.handleSearch(selectedKeys, confirm)}
          icon="search"
          size="small"
          style={{ width: 90, marginRight: 8 }}
        >
          Search
        </Button>
        <Button onClick={() => this.handleReset(clearFilters)} size="small" style={{ width: 90 }}>
          Reset
        </Button>
      </div>
    ),
    filterIcon: filtered => (
      <Icon type="search" style={{ color: filtered ? '#1890ff' : undefined }} />
    ),
    onFilter: (value, record) => record[dataIndex]
      .toString()
      .toLowerCase()
      .includes(value.toLowerCase()),
    onFilterDropdownVisibleChange: (visible) => {
      if (visible) {
        setTimeout(() => this.searchInput.select());
      }
    },
  });

  handleSearch = (selectedKeys, confirm) => {
    confirm();
    this.setState({ keyword: selectedKeys[0] });
  };

  handleReset = (clearFilters) => {
    clearFilters();
    this.setState({ keyword: '' });
  };

  render() {
    return (
      <div className={style.searchWrapper} ref={this.setWrapperRef}>
        <Col span={24} className={style.innerWrapper}>
          <Input
            id="analyticsSearch"
            placeholder= {this.props.placeholder}
            className={style.input}
            onPressEnter={e => this.handleSubmit(e.target.value)}
            autoComplete="off"
          />
          <Button
            type="primary"
            shape="circle"
            size="default"
            className={style.searchIcon}
            onClick={this.handleSearchClick}
          >
            <img src={`${publicRuntimeConfig.asset}/static/dist/images/search.svg`} alt="search icon" />
          </Button>
          <Icon
            type="loading"
            theme="outlined"
            className={`anticon-spin ${style.spinner} ${!this.state.submit && this.state.autocompleteLoading ? style.show : style.hide}`}
          />
        </Col>
      </div>
    );
  }
}

export default AnalyticsSearch;
