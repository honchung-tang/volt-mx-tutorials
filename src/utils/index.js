import _startCase from 'lodash/startCase';
import _toLower from 'lodash/toLower';
import moment from 'moment';

/**
 *  Truncates the string.
 * @private
 * @param {string} string - The string to truncate.
 * @param {number} length - The length of truncated string.
 * @returns {string} Returns truncated `string`.
 */

export const truncate = (string, length) => {
  if (typeof length !== 'number' || typeof string !== 'string') return string;
  if (string.length <= length) return string;
  if (length <= 3) return `${string.slice(0, length)}...`;
  return `${string.slice(0, length - 3)}...`;
};

/**
 *  Truncates the string without cutting words.
 * @private
 * @param {string} string - The string to truncate.
 * @param {number} length - The length of truncated string.
 * @returns {string} Returns truncated `string`.
 */

export const truncateAdvance = (string, length) => {
  if (typeof length !== 'number' || typeof string !== 'string') return string;
  if (string.length <= length) return string;
  return `${string.replace(new RegExp(`^(.{${length}}[^\\s]*).*`), '$1')}`;
};

/**
 *  Capitalize First Letter of the string.
 * @private
 * @param {string} string - The string whose first letter to be capitalized.
 * @returns {string} Returns string whose first letter is capital.
 */

export const capitalizeFirstLetter = (string) => {
  if (!string) {
    return null;
  }
  return _startCase(_toLower(string));
};

/**
 *  Get month name from the timestamp.
 * @private
 * @param {number} timestamp - The timestamp which is going to converted.
 * @returns {string} Returns month name from timestamp passed.
 */

export const getMonthNameFromTimestamp = (timestamp) => {
  const date = new Date(timestamp);
  const monthName = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];
  return monthName[date.getMonth()];
};

/**
 *  Get device os platform.
 * @private
 * @returns {string} Returns name of platform.
 */

let os = null;
if (process.browser) {
  const { userAgent, platform } = window.navigator;
  const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
  const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'];
  const iosPlatforms = ['iPhone', 'iPad', 'iPod', 'iPhone Simulator', 'iPod Simulator', 'iPad Simulator'];
  if (macosPlatforms.indexOf(platform) !== -1) {
    os = 'Mac OS';
  } else if (iosPlatforms.indexOf(platform) !== -1) {
    os = 'iOS';
  } else if (windowsPlatforms.indexOf(platform) !== -1) {
    os = 'Windows';
  } else if (/Android/.test(userAgent)) {
    os = 'Android';
  } else if (!os && /Linux/.test(platform)) {
    os = 'Linux';
  }
}

export const getOS = os;

/**
 * Load a script.
 * @private
 * @param src of the file
 */

export const loadScript = (src) => {
  const tag = document.createElement('script');
  tag.src = src;
  tag.async = true;
  document.body.appendChild(tag);
};

export const validateData = (d) => {
  if (d) {
    if (typeof (d) === 'string' && d !== '') {
      return true;
    }
    if (typeof (d) === 'object' && Object.keys(d).length > 0) {
      return true;
    }
  }
  return false;
};

const escapeRegExp = string => string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');

/**
 * Replace all occurences of string with other.
 *
 * @private
 *
 * @param str, String to be replaced on.
 * @param term, term to be replaced.
 * @param replacement, String to be replaced with.
 */
export const replaceAll = (str, term, replacement) => str.replace(new RegExp(escapeRegExp(term), 'g'), replacement);

/**
 * Convert number into two digits.
 *
 * @private
 *
 * @param number, Number to be converted.
 *
 * @return String, Number in two digits.
 */
function numberToTwoDigits(number) {
  return number > 9 ? `${number}` : `0${number}`;
}

/**
 * Convert seconds into HH:mm:ss time format.
 *
 * @private
 *
 * @param second, Number of seconds.
 *
 * @return String, time in the HH:mm:ss format.
 */
export const secondsToTime = (second) => {
  const seconds = parseInt(second, 10);
  const momentDuration = moment.duration(seconds, 'seconds');
  const hours = numberToTwoDigits(Math.floor(momentDuration.asHours()));
  const mins = numberToTwoDigits(momentDuration.minutes());
  const secs = numberToTwoDigits(momentDuration.seconds());
  return `${hours}:${mins}:${secs}`;
};
