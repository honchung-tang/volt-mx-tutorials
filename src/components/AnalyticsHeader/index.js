import React from 'react';
import Col from 'antd/lib/col';
import Row from 'antd/lib/row';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import style from './style.scss';
import AnalyticsSearch from '../AnalyticsSearch';

export const AnalyticsHeader = ({
  title, subTitle, filterData, placeholder, search,
}) => (
  <Row className={style.siteHeader}>
    <Col span={12}>
      <h1 className={style.headerTitle}>{title}</h1>
      <h2 className={style.headersubTitle}>{subTitle}</h2>
    </Col>
    {
      search
        ? (
          <Col span={12}>
            <AnalyticsSearch
              filterData={filterData}
              placeholder={placeholder}
            />
          </Col>
        )
        : null
    }
  </Row>
);

AnalyticsHeader.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
  search: PropTypes.bool,
  filterData: PropTypes.func,
  placeholder: PropTypes.string,
};

AnalyticsHeader.defaultProps = {
  title: 'HCL Volt MX Tutorials',
  subTitle: 'New Assets, New Possibilities',
  search: false,
  filterData: () => {},
  placeholder: '',
};

const mapStateToProps = ({ marketplace }) => ({ marketplace });

export default connect(mapStateToProps)(AnalyticsHeader);
