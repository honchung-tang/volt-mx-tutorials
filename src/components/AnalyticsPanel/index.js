import React from 'react';
import Col from 'antd/lib/col';
import Row from 'antd/lib/row';
import PropTypes from 'prop-types';
import style from './style.scss';
import { truncateAdvance } from '../../utils';
import getConfig from 'next/config';
const { publicRuntimeConfig } = getConfig();

const AnalyticsPanel = ({ data, title, className }) => (
  <Row className={`${style.base} ${className}`}>
    <Col className={style.title}>
      <h2>
        TOP 5
      </h2>
    </Col>
    <Col>
      <h2>
        {title}
      </h2>
    </Col>
    <Col span={24} className={style.wrapper}>
      {
        data && data.length > 0 ? data.map(item => (
          <Row className={style.itemWrapper}>
            <Col span={16}>
              {
                truncateAdvance(item.title, 30)
              }
            </Col>
            <Col span={4}>
              <img
                src={`${publicRuntimeConfig.asset}/static/dist/images/icons/like.svg`}
                alt="icon"
                className={style.statsIcon}
              />
              {
                item.likes
              }
            </Col>
            <Col span={4}>
              <img
                src={`${publicRuntimeConfig.asset}/static/dist/images/icons/downloadBlue.png`}
                alt="icon"
                className={style.statsIcon}
              />
              {
                item.downloads
              }
            </Col>
          </Row>
        )) : 'No Data found for current Selected Dates'
      }
    </Col>
  </Row>
);

AnalyticsPanel.propTypes = {
  title: PropTypes.string,
  data: PropTypes.array,
};

AnalyticsPanel.defaultProps = {
  title: '',
  data: [],
};

export default AnalyticsPanel;
