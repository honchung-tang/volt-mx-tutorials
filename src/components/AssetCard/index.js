import React, { Component } from 'react';
import Icon from 'antd/lib/icon';
import Col from 'antd/lib/col';
import Row from 'antd/lib/row';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { Base64 } from 'js-base64';
import { connect } from 'react-redux';
import style from './style.scss';
import { truncate, validateData } from '../../utils';
import KonyButton from '../KonyButton';
import { getUrl } from '../../utils/initialize';
import getConfig from 'next/config';
const { publicRuntimeConfig } = getConfig();

class AssetCard extends Component {
  state = {
    itemPath: '#',
    imgOpacity: 1,
    loaded: false,
    loaderDisplay: 'none',
    imagePath: this.props.asset.Logo,
    hover: false,
    query: this.props.keyword ? `search=${this.props.keyword}` : '',
    isViz: this.props.marketplace.isVizWeb,
  };

  isPreviewPresent = validateData(this.props.asset.AppPreviewCode);

  authorURL = () => {
    const { AuthorName, AuthorUID } = this.props.asset;
    const { isVizWeb } = this.props.marketplace;
    const encodedAuthor = Base64.encode(`${AuthorName}::${AuthorUID}`);
    return getUrl(`/author/${encodedAuthor}`, null, isVizWeb);
  };

  itemHover = (e) => {
    e.preventDefault();
    this.setState({
      hover: true,
    });
    if (this.state.imagePath === this.props.asset.Logo && !this.state.loaded) {
      this.setState({
        imgOpacity: 0.5,
        loaderDisplay: 'block',
        itemPath: this.props.asset.Alias,
        imagePath: this.props.asset.OriginalLogo,
      });
    }
  };

  hideLoader = (e) => {
    e.preventDefault();
    if (this.state.hover) {
      this.setState({
        loaderDisplay: 'none',
        imgOpacity: 1,
        loaded: true,
      });
    }
  };

  resetImage = () => {
    this.setState({
      imagePath: this.props.asset.Logo,
      loaded: false,
      loaderDisplay: 'none',
      imgOpacity: 1,
      hover: false,
    });
  };

  assetClick = () => {
    window.dataLayer.push({
      assetTitle: this.props.asset.Title,
      assetAlias: this.props.asset.Alias,
      event: 'assetClick',
    });
  };

  render() {
    const { asset, marketplace } = this.props;
    const { isVizWeb } = marketplace;
    const {
      Title, Alias, AuthorName, Description, Likes,
    } = asset;
    const {
      query, itemPath, imagePath, imgOpacity, loaderDisplay,
    } = this.state;
    return (
      <div className={style.card}>
        <div
          className={style.cover}
          onMouseOver={this.itemHover}
          onFocus={this.itemHover}
          onMouseLeave={this.resetImage}
          onLoad={this.hideLoader}
          onBlur={this.resetImage}
        >
          <Link href={getUrl(itemPath, query, isVizWeb)}>
            <a onClick={this.assetClick} aria-label={`label-${Title}`}>
              <img
                tabIndex="-1"
                src={imagePath}
                alt={`logo-${Title}`}
                style={{ opacity: imgOpacity }}
                className={style.image}
              />
              <div
                className={style.loadingImage}
                style={{ display: loaderDisplay }}
              >
                <img
                  src={`${publicRuntimeConfig.asset}/static/dist/images/loader.svg`}
                  alt="Loader"
                  height={80}
                  width={80}
                />
              </div>
            </a>
          </Link>
        </div>
        <Col className={style.meta}>
          <Row type="flex" className={style.flexRow}>
            <Col
              xs={this.isPreviewPresent ? 15 : 22}
              sm={this.isPreviewPresent ? 18 : 22}
              md={24}
              lg={24}
              className={style.fullHeight}
            >
              <div className={style.fullWidth}>
                <Link href={getUrl(Alias, query, isVizWeb)}>
                  <h4 className={style.title} onClick={this.assetClick}>
                    {
                      truncate(Title, this.isPreviewPresent ? 24 : 26)
                    }
                  </h4>
                </Link>
                <h5 className={style.author}>
                  By:
                  {' '}
                  <Link href={this.authorURL()}>{AuthorName}</Link>
                </h5>
                <Col
                  xs={24}
                  sm={24}
                  md={0}
                  lg={0}
                  className={style.desc}
                >
                  <Link href={getUrl(Alias, query, isVizWeb)}>
                    <span>
                      {
                        truncate(Description, this.isPreviewPresent ? 44 : 55)
                      }
                    </span>
                  </Link>
                </Col>
                <Col
                  lg={24}
                  md={24}
                  xs={0}
                  sm={0}
                  className={style.desc}
                >
                  <Link href={getUrl(Alias, query, isVizWeb)}>
                    <span>
                      {
                        truncate(Description, 70)
                      }
                    </span>
                  </Link>
                </Col>
                <Col className={style.stats} type="flex" md={24} xs={0}>
                  <a href={getUrl(Alias, query, isVizWeb)} aria-label={`label-${Title}`}>
                    <span>
                      <img
                        src={`${publicRuntimeConfig.asset}/static/dist/images/icons/like.svg`}
                        alt={`like-${Title}`}
                        className={style.likeIcon}
                      />
                      {Likes}
                    </span>
                    <span>
                      <img
                        src={`${publicRuntimeConfig.asset}/static/dist/images/icons/download.svg`}
                        alt={`download-${Title}`}
                        className={style.downloadIcon}
                      />
                      {asset.Downloads}
                    </span>
                  </a>
                </Col>
              </div>
            </Col>
            <Col
              lg={0}
              md={0}
              xs={this.isPreviewPresent ? 9 : 2}
              sm={this.isPreviewPresent ? 6 : 2}
              className={style.previewButtonWrapper}
            >
              {this.isPreviewPresent && marketplace?.userType !== 'viewer' ? (
                <KonyButton
                  type="appPreview"
                  title="PREVIEW"
                  href={`https://www.kony.com/marketplace/app_preview?apptolaunch=${
                    asset.AppPreviewCode
                    }`}
                />
              ) : null}
              <Link href={getUrl(asset.Alias, query, isVizWeb)}>
                <a onClick={this.assetClick} aria-label={`label-${Title}`}>
                  <Icon
                    type="right"
                    theme="outlined"
                    className={style.angleRight}
                  />
                </a>
              </Link>
            </Col>
          </Row>
        </Col>
      </div>
    );
  }
}

AssetCard.propTypes = {
  asset: PropTypes.object,
};

AssetCard.defaultProps = {
  asset: {},
};

const mapStateToProps = ({ marketplace }) => ({ marketplace });

export default connect(mapStateToProps)(AssetCard);
