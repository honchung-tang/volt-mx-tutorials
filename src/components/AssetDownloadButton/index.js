import React, { Component } from 'react';
import Modal from 'antd/lib/modal';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import style from './style.scss';
import KonyButton from '../KonyButton';
import VizButton from './VizButton';
import ComingSoon from './comingSoon';
import ContactUs from './contactUs';
import CoreAsset from './CoreAsset';
import SuccessModal from '../SuccessModal';
import KonyLoader from '../KonyLoader';
import { instance as axios } from '../../utils/initialize';
import { validateData } from '../../utils';

class AssetDownloadButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false,
      modalError: false,
      modalMessage: "Thanks for reaching out!",
      modalLoading: false,
      isNotified: this.props.isNotified,
      isViz: this.props.marketplace.isVizWeb,
      isVizApp: this.props.marketplace.isVizApp,
      oauthURL: `/oauth/login?destination=${this.props.url.asPath || '/'}`,
      validDownloadURLPresent: validateData(this.props.fileURL),
    };
  }

  handleClose = () => {
    this.setState({
      showModal: false,
    });
  };

  success = (message) => {
    this.setState({
      modalLoading: false,
      showModal: true,
      modalError: false,
      modalMessage: message || "Thanks for reaching out!",
      isNotified: true,
    });
  };

  error = (message) => {
    this.setState({
      modalLoading: false,
      showModal: true,
      modalError: true,
      modalMessage: message || "Something went wrong. Please try again later.",
      visible: true,
    });
  };

  getDownloadButtonURL = () => {
    if (this.props.marketplace.isLogin) {
      return `${this.props.fileURL}?agent=web`;
    }
    return this.state.oauthURL;
  }

  handleClick = (type) => {
    if (!this.props.marketplace.isLogin && !this.state.isViz) {
      return null;
    }
    if (type === "contact" || type === "notify") {
      this.handleContactSubmit(type);
    }
    if (type === "download") {
      setTimeout(function () {
        window.dataLayer.push({
          'assetTitle': this.props.title,
          'event': 'webDownload'
        })
      }.bind(this), 1000);
    }
  }

  handleContactSubmit = async (type) => {
    this.setState({
      modalLoading: true,
    });
    const data = new FormData();
    data.append('title', this.props.title);
    data.append('version', this.props.Version);
    data.append('assetID', this.props.itemId);
    data.append('notify', type === 'notify');

    const headers = {
      'Content-Type': 'application/x-www-form-urlencoded',
    };
    axios({
      method: 'post',
      url: 'api/v1_0/marketplace/post/contact',
      data,
      config: {
        headers,
      },
    })
      .then(res => {
        this.success(res.data.message);
      })
      .catch(error => this.error());
  }
  renderButton() {
    const { isViz, isVizApp, validDownloadURLPresent, isNotified, oauthURL } = this.state;
    const { isLogin } = this.props.marketplace;

    if (!validDownloadURLPresent) {
      if (this.props.termMapName === "coming_soon") {
        return <ComingSoon hideNotify={isVizApp} disabled={isNotified} href={oauthURL} isLogin={isLogin || isViz} submit={() => this.handleClick("notify")} />;
      }
      if (this.props.isCoreAsset) {
        return <CoreAsset isLogin={isLogin || isViz} />;
      }
      if (!isVizApp) {
        return <ContactUs href={oauthURL} disabled={isNotified} isLogin={isLogin || isViz} submit={() => this.handleClick("contact")} />;
      }
      return null;
    };

    if (this.props.isCoreAsset) {
      return <CoreAsset isLogin={isLogin || isViz} />;
    }

    if (this.props.isContactUs && !isVizApp) {
      return <ContactUs href={oauthURL} disabled={isNotified} isLogin={isLogin || isViz} submit={() => this.handleClick("contact")} />;
    }

    if (this.props.isMobile) {
      return null;
    }

    if (this.props.termMapName === "coming_soon") {
      return <ComingSoon hideNotify={isVizApp} disabled={isNotified} href={oauthURL} isLogin={isLogin || isViz} submit={() => this.handleClick("notify")} />;
    }

    if (isViz) {
      return (<VizButton {...this.props} disabled={!validDownloadURLPresent} />);
    }
    return (
      <KonyButton
        title={isLogin ? "Download" : "Login to Download"}
        type="download"
        color="blue"
        className={`${style.assetButtons} ${isLogin ? style.pushLeft : null}`}
        href={this.getDownloadButtonURL()}
        disabled={!validDownloadURLPresent}
        onClick={() => this.handleClick("download")}
      />
    );
  }

  render() {
    return (
      <React.Fragment>
        {this.renderButton()}
        {this.state.modalLoading ? (<KonyLoader style={{ opacity: '0.6' }} />) : null}
        <Modal
          centered
          visible={this.state.showModal}
          onCancel={e => this.handleClose(e)}
          onOk={e => this.handleClose(e)}
          closable={false}
          footer={null}
          className={style.successModal}
        >
          <SuccessModal
            error={this.state.modalError}
            handleClick={e => this.handleClose(e)}
            message={this.state.modalMessage}
          />
        </Modal>
      </React.Fragment>
    );
  }
}
AssetDownloadButton.propTypes = {
  fileURL: PropTypes.string,
};

AssetDownloadButton.defaultProps = {
  fileURL: '',
};

const mapStateToProps = ({ marketplace }) => ({ marketplace });

export default connect(mapStateToProps)(AssetDownloadButton);
