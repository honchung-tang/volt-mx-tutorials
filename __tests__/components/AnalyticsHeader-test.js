import React from 'react';
import { AnalyticsHeader } from '../../src/components/AnalyticsHeader';
import { shallow } from '../../src/utils/enzyme';

describe('AnalyticsHeader component test suite', () => {
  it('render JSX without search', () => {
    const wrappedComponent = shallow(<AnalyticsHeader title="Analytics Header" subTitle="Analytics Sub Title" filterData={() => {}} placeholder="placeholder" />);
    expect(wrappedComponent).toMatchSnapshot();
  });
  it('render JSX with search', () => {
    const wrappedComponent = shallow(<AnalyticsHeader title="Analytics Header" subTitle="Analytics Sub Title" filterData={() => {}} placeholder="placeholder" search />);
    expect(wrappedComponent).toMatchSnapshot();
  });
});
