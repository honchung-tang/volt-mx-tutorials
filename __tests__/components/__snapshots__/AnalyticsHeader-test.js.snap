// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsHeader component test suite render JSX with search 1`] = `
<Row
  className="siteHeader"
  gutter={0}
>
  <Col
    span={12}
  >
    <h1
      className="headerTitle"
    >
      Analytics Header
    </h1>
    <h2
      className="headersubTitle"
    >
      Analytics Sub Title
    </h2>
  </Col>
  <Col
    span={12}
  >
    <AnalyticsSearch
      filterData={[Function]}
      placeholder="placeholder"
    />
  </Col>
</Row>
`;

exports[`AnalyticsHeader component test suite render JSX without search 1`] = `
<Row
  className="siteHeader"
  gutter={0}
>
  <Col
    span={12}
  >
    <h1
      className="headerTitle"
    >
      Analytics Header
    </h1>
    <h2
      className="headersubTitle"
    >
      Analytics Sub Title
    </h2>
  </Col>
</Row>
`;
